ecl/hqlcpp/hqlcfilter.cpp

```diff
@@ -328,6 +342,36 @@ void CppFilterExtractor::createStringSet(BuildCtx & ctx, const char * target, un
     }
 }
 
+bool CppFilterExtractor::buildSubRange(BuildCtx & ctx, IHqlExpression * range, CHqlBoundExpr & bound)
+{
+    if (!createValueSets)
+        return false;
+
+    if (!range)
+        return false;
+
+    IHqlExpression * limit;
+    switch (range->getOperator())
+    {
+    case no_rangeto:
+        limit = range->queryChild(0);
+        break;
+    case no_range:
+        assertex(matchesConstValue(range->queryChild(0), 1));
+        limit = range->queryChild(1);
+        break;
+    case no_constant:
+        limit = range;
+        assertex(matchesConstValue(range, 1));
```

Comment:
I don't understand this line

Reply:
I think it means that we should only be here with leading substrings, including leading single-char substrings. Which makes sense (and this code appears to have moved from elsewhere)